src/codegen/mod.rs

```diff
@@ -2009,23 +2009,27 @@ impl CodeGenerator for CompInfo {
         let mut needs_default_impl = false;
         let mut needs_debug_impl = false;
         let mut needs_partialeq_impl = false;
+        let mut needs_packed_wrapper = false;
         if let Some(comment) = item.comment(ctx) {
             attributes.push(attributes::doc(comment));
         }
+
+        // We can't specify both packed(N) and align(N), but the packed()
+        // should be redundant in this case.
         if packed && !is_opaque {
             let n = layout.map_or(1, |l| l.align);
             assert!(ctx.options().rust_features().repr_packed_n || n == 1);
-            let packed_repr = if n == 1 {
+            if explicit_align.is_some() {
+                needs_packed_wrapper = true;
+            }
+            let packed_repr = if n == 1 || needs_packed_wrapper {
                 "packed".to_string()
             } else {
                 format!("packed({})", n)
             };
             attributes.push(attributes::repr_list(&["C", &packed_repr]));
         } else {
             attributes.push(attributes::repr("C"));
-        }
-
-        if ctx.options().rust_features().repr_align {
```

Comment:
Why removing the feature-check for `repr(align)`?

Reply:
accidently happened during multiple revision 😂